maven-enforcer-plugin/src/main/java/org/apache/maven/plugins/enforcer/EnforceMojo.java

```diff
@@ -134,9 +147,7 @@ public void execute() throws MojoExecutionException {
         Log log = this.getLog();
 
         PluginParameterExpressionEvaluator evaluator = new PluginParameterExpressionEvaluator(session, mojoExecution);
-        if (commandLineRules != null && commandLineRules.length > 0) {
-            this.rules = createRulesFromCommandLineOptions();
-        }
+        this.rules = determineRulesToExecute();
```

Comment:
I am not the biggest fan of this re-assignment to the instance field, but this is the current way of doing it, so I left it for now.

I could potentially just create a local variable with the updated list and leave the fields be. This could be however considered a breaking change since the `getRules()` would then be returning different list than before (updated vs original).

Reply:
technically plugin doesn't need getters and setters - I see that `getRules` is not used

Reply:
Yeah, maybe I am too paranoid. I was thinking if someone is doing e.g. `new EnforcerMojo().getRules()` then removing the getter would break that. Good question is how much sense does that even make and if this is anything that we should care about.

If you agree I would remove the getter then.